### PR TITLE
Bubble Sort

### PR DESCRIPTION
**Pull request by BigWizzo**
# Ruby: Bubble Sort
> Implementing the bubble sort algorithm

### CODE REVIEW

**Review by kitpao — CHANGES_REQUESTED** [review]
.

File: README.md [hunk 1]
```diff
@@ -2,64 +2,51 @@
...
+- Text Editor
 
 ### Usage
-
-### Run tests
-
-### Deployment
-
-
+- On the terminal run ruby bubble_sort.rb
```
Comment (kitpao) [inline README.md:29]: - [ ] **Optional:** to use formatting for code, you can surround the code with backquotes `like this` 
  ```
  `like this`
  ```

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  x = 0
+  length = array.length
+  while x < length - 1
+    if !array[x + 1].nil? && array[x] > array[x + 1]
+      array[x], array[x + 1] = array[x + 1], array[x]
+      x = 0
+    end
+    x += 1
+  end
+  p array
```
Comment (kitpao) [inline bubble_sort.rb]: - [x] Notice that normal methods only return the needed value, and if the user wants to print, they do it on their own. So your method only has to return the value. Please, delete that `p` and just return `array` (same in line 29)

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      end
+    end
+    x += 1
+  end
+  p array
+end
+
+bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
```
Comment (kitpao) [inline bubble_sort.rb]: - [x] Please, delete all debugging code that is outside of the definition of the methods. (same in line 14)
image

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  length = array.length
+  while x < length - 1
+    if !array[x + 1].nil? && array[x] > array[x + 1]
+      array[x], array[x + 1] = array[x + 1], array[x]
+      x = 0
+    end
+    x += 1
+  end
+  p array
+end
```
Comment (kitpao) [inline bubble_sort.rb]: - [x] If you test this array `p bubble_sort([10,9,8,7,6,5,4,3,2,1])` it gives [9, 1, 2, 3, 4, 5, 6, 7, 8, 10], please check your logic

  **Hint:** About your logic, please notice that the order of the comparison in bubble sort is like this:
SWYiwVcMCbdsToyWuAZscP5obiz1-3762u3wq4
Both your methods process the array like this with each change:
image

👉 And in bubble sort, the bigger number should be accommodated in the last position first

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello @BigWizzo and @Kingobaino1 

🌟 Nice job setting your linters for this new module

📝 You have an issue with your logic, please check below to see an explanation and required changes 🔽 

- [x] Although you have a pull request message, try to make it more descriptive, at least put the 2 different methods and their differences

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me in slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate if can leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😛 

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**Comment by Kingobaino1** [comment]
**In this pull request, we**
- created bubble_sort method
  - which sorts numbers in array in ascending order
- created bubble_sort_by method
  - which sorts strings in an array by their length (i.e., the string with the smallest length comes first and the string with the highest length comes last.)
  - the method is run by a block outside the method.

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Great job guys your bubble sort project is complete.

now you can merge your project is approved, :warning: but we have an issue here you are not using git-flow please remember it next time.

## Happy Coding :tophat:

**Pull request merged**